base/libgit2/reference.jl

```diff
@@ -30,24 +30,49 @@ function isorphan(repo::GitRepo)
     r != 0
 end
 
+"""
+    LibGit2.head(repo::GitRepo) -> GitReference
+
+Returns a `GitReference` to the current HEAD of `repo`.
+"""
 function head(repo::GitRepo)
     head_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
     @check ccall((:git_repository_head, :libgit2), Cint,
                   (Ptr{Ptr{Void}}, Ptr{Void}), head_ptr_ptr, repo.ptr)
     return GitReference(repo, head_ptr_ptr[])
 end
 
+"""
+    LibGit2.shortname(ref::GitReference)
+
+Returns a shortened version of the name of `ref` that's
+"human-readable".
+"""
 function shortname(ref::GitReference)
     isempty(ref) && return ""
     name_ptr = ccall((:git_reference_shorthand, :libgit2), Cstring, (Ptr{Void},), ref.ptr)
     name_ptr == C_NULL && return ""
     return unsafe_string(name_ptr)
 end
 
+"""
+    LibGit2.reftype(ref::GitReference) -> Cint
+
+Returns a `Cint` corresponding to the type of `ref`:
+  * `0` if the reference is invalid
+  * `1` if the reference is an object id
+  * `2` if the reference is symbolic
+"""
 function reftype(ref::GitReference)
     return ccall((:git_reference_type, :libgit2), Cint, (Ptr{Void},), ref.ptr)
 end
 
+"""
+    LibGit2.fullname(ref::GitReference)
+
+Return the name of the reference pointed to by the
+symbolic reference `ref`.
```

Comment:
Add something to the effect of "If `ref` is not a symbolic reference [or whatever else could cause the ccall to return a null pointer?] then returns the empty string."